### DISCUSSION TITLE
Confused Calibration 1 Error

### DISCUSSION THREAD

**Discussion by Lynd** [Navid200/xDrip#30] [Dexcom]
When calibrating restarted G6 sensors.  I often receive an error message "Confused Calibration 1" at which time the transmitter stops sending readings. 

For example, a couple of days ago, I restarted a 10 day old sensor using the no-code method.  Shortly after the first reading, I entered the initial "double calibration" values.   Then, I  received a "confused calibration 1" error that caused the readings to completely stop  So, I stopped and restarted the sensor, following [Navid200's guides](https://github.com/Navid200/xDrip/wiki/Restart-G6-Sensor). 

On restart, the readings were very high, so  I used  the original code and calibrated the sensor within about 12 percent of the reading, and then an hour later I calibrated again about 10 percent of the reading (At the time I was entering the calibrations, my readings were only changing by 3-5 points). A few minutes later, "confused calibration 1" appeared and  xDrip did not receive any readings after that.  

Note that:  1) I disabled "double calibration" mode when using the original code method; 2) I'm using the latest xDrip build; and 3) I experienced the same "confused calibration 1" errors using different transmitters and sensors.

**Comment by Navid200** [comment]
>an hour later I calibrated again about 10 percent of the reading 

Would you please state what the values were and what you calibrated to?  

<br/>

>my readings were only changing by 3-5 points

Just to be sure, would you please confirm you mean 3-5 mg/dL?

**Reply by Lynd** [reply]
I haven't been able to figure out how to view the history of the calibration values that I entered, so not sure about all of them. Do you know whether there is a log file for the calibration values entered?   Anyway, this is what I recall:

 The last reading value from yesterday was 206 mg/dl and when I entered calibrations of 185 mg/dl and received the "confused calibration 1".   I restarted again this morning, and,  I was able to enter two separate calibrations (about 30 minutes apart) without error.  Then, after another 25 minutes, I calibrated again with a value of  145 mg/dl when the xDrip reading was 156 mg/dl. This caused a "confused calibration 1" error, which would not go away.

So, in summary the values that caused the errors were:   185 calibration when xDrip showed 206  and 145 calibration when xDript showed 156.

**Reply by Navid200** [reply]
Event logs will show the calibration values.  Of course, the logs are not kept forever.  But, they last more than 24 hours.

I wouldn't worry about calibrating to be perfectly accurate after a restart.  In 24 hours, the error will disappear.  If you calibrate it perfectly, tomorrow, your readings will be low and you may need to calibrate high then.

But, I am surprised about the error you got today (145 versus 156).

I usually, calibrate only twice after a restart.  I am going to think about how to edit my guide to warn reader about the possibility of this issue you have raised.  Thanks for that.

**Reply by Lynd** [reply]
Thanks. So, would your suggested solution be to only calibrate twice and keep the calibrations to under 20 percent?  I'm pretty sure that's what I did yesterday when I got the error, but will check the logs. 

> If you calibrate it perfectly, tomorrow, your readings will be low and you may need to calibrate high then.

  Do the values keep adjusting lower after the following day or do they stabilize?

**Reply by Navid200** [reply]
I will probably add that if you have to calibrate a third time, wait 45 minutes after the second calibration.

Assuming you don't need to calibrate with a new sensor, after a restart with the original code, if you wait 24 hours, you will be very close.  After 48 hours, the error becomes negligible.

**Reply by Lynd** [reply]
Thanks again.  So, in order to account for the high readings between hour 2 and 24, how close do you initially calibrate a restarted sensor?

**Reply by Navid200** [reply]
I wouldn't worry too much if I was closer than 13mg/dL or 0.7mmol/L.
But, I don't use a closed-loop system.  So, I am aware that it is a little bit high and won't overdo insulin until tomorrow.

**Reply by Lynd** [reply]
That sounds reasonable,  With the transmitters that send raw data, can one use the Dexcom algorithm and calculate the correct readings.

**Reply by Navid200** [reply]
If you meant if one can use the xDrip algorithm ..., the answer is yes by using the non-native mode.  However, you will be giving up factory calibration.
https://github.com/Navid200/xDrip/wiki/Native-Algorithm
